src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts

```diff
@@ -91,15 +91,13 @@ function getComponentStartComment(node: Node): StartComponentComment | undefined
           return {
             node: node as Comment,
             circuitId,
-            rendererId: rendererId,
-            componentId: componentId,
+            rendererId: Number.parseInt(rendererId),
+            componentId: Number.parseInt(componentId),
```

Comment:
This was removed in the previous commit and caused the regression. We *do* have a test for mixed string and int already which is how we found the break.

Reply:
This is weird, it should not be needed.

https://github.com/aspnet/AspNetCore/blob/7b56439ca1cbc384f1e9648f4ae983d3b39119d8/src/Components/Server/src/Circuits/CircuitPrerenderer.cs#L185-L199

Reply:
we have tests for it bruh. That verify that can parse when rendererId/circuitId is a string 😆 

The end-tag version does a conversion as well.

Reply:
Which tests?

Reply:
https://github.com/aspnet/AspNetCore/blob/master/src/Components/Server/test/Circuits/CircuitPrerendererTest.cs#L23

Reply:
I'm happy to fix this in either direction, but something needs to change.

Reply:
If the tests are JS tests, they aren't up to date.